small branch window size crashes

```
[ERROR] panic: PanicInfo { payload: Any, message: Some(Trying to access position outside the buffer: x=45, y=13, area=Rect { x: 0, y: 0, width: 45, height: 46 }), location
: Location { file: "/Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/buffer.rs", line: 217, col: 9 } }
trace:
   0: backtrace::backtrace::libunwind::trace
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.53/src/backtrace/libunwind.rs:100:5
      backtrace::backtrace::trace_unsynchronized
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.53/src/backtrace/mod.rs:66:5
   1: backtrace::backtrace::trace
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.53/src/backtrace/mod.rs:53:14
   2: backtrace::capture::Backtrace::create
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.53/src/capture.rs:176:9
   3: backtrace::capture::Backtrace::new
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.53/src/capture.rs:140:22
   4: gitui::set_panic_handlers::{{closure}}
             at src/main.rs:329:25
   5: std::panicking::rust_panic_with_hook
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:573:17
   6: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:476:9
   7: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:153:18
   8: rust_begin_unwind
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475:5
   9: std::panicking::begin_panic_fmt
             at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:429:5
  10: tui::buffer::Buffer::index_of
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/buffer.rs:217:9
  11: tui::buffer::Buffer::get_mut
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/buffer.rs:184:17
  12: <tui::widgets::clear::Clear as tui::widgets::Widget>::render
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/widgets/clear.rs:32:17
  13: tui::terminal::Frame<B>::render_widget
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/terminal.rs:103:9
  14: <gitui::components::select_branch::SelectBranchComponent as gitui::components::DrawableComponent>::draw
             at src/components/select_branch.rs:62:13
  15: gitui::app::App::draw_popups
             at src/app.rs:626:9
  16: gitui::app::App::draw
             at src/app.rs:205:9
  17: gitui::draw::{{closure}}
             at src/main.rs:190:25
  18: tui::terminal::Terminal<B>::draw
             at /Users/stephan/.cargo/registry/src/github.com-1ecc6299db9ec823/tui-0.12.0/src/terminal.rs:259:9
  19: gitui::draw
             at src/main.rs:189:5
  20: gitui::main
             at src/main.rs:155:13
```

--- a/src/cmdbar.rs
+++ b/src/cmdbar.rs
@@ -186,7 +186,7 @@ impl CommandBar {
                 r.width.saturating_sub(MORE_WIDTH),
                 r.y + r.height.saturating_sub(1),
                 MORE_WIDTH.min(r.width),
-                1,
+                1.min(r.height),
             );
 
             f.render_widget(

--- a/src/components/help.rs
+++ b/src/components/help.rs
@@ -3,6 +3,7 @@ use super::{
     DrawableComponent,
 };
 use crate::{keys::SharedKeyConfig, strings, ui, version::Version};
+use anyhow::Result;
 use asyncgit::hash;
 use crossterm::event::Event;
 use itertools::Itertools;
@@ -15,8 +16,6 @@ use tui::{
     widgets::{Block, BorderType, Borders, Clear, Paragraph},
     Frame,
 };
-
-use anyhow::Result;
 use ui::style::SharedTheme;
 
 ///

--- a/src/components/select_branch.rs
+++ b/src/components/select_branch.rs
@@ -17,7 +17,7 @@ use crossterm::event::Event;
 use std::{cmp, convert::TryFrom};
 use tui::{
     backend::Backend,
-    layout::{Alignment, Constraint, Direction, Layout, Rect},
+    layout::{Alignment, Rect},
     text::{Span, Spans, Text},
     widgets::{Block, BorderType, Borders, Clear, Paragraph},
     Frame,
@@ -40,10 +40,8 @@ impl DrawableComponent for SelectBranchComponent {
     fn draw<B: Backend>(
         &self,
         f: &mut Frame<B>,
-        _rect: Rect,
+        rect: Rect,
     ) -> Result<()> {
-        // Render a scrolllist of branches inside a box
-
         if self.visible {
             const PERCENT_SIZE: (u16, u16) = (60, 25);
             const MIN_SIZE: (u16, u16) = (50, 20);
@@ -54,6 +52,7 @@ impl DrawableComponent for SelectBranchComponent {
                 f.size(),
             );
             let area = ui::rect_min(MIN_SIZE.0, MIN_SIZE.1, area);
+            let area = area.intersection(rect);
 
             let scroll_threshold = area.height / 3;
             let scroll =
@@ -61,29 +60,18 @@ impl DrawableComponent for SelectBranchComponent {
 
             f.render_widget(Clear, area);
             f.render_widget(
-                Block::default()
-                    .title(strings::SELECT_BRANCH_POPUP_MSG)
-                    .borders(Borders::ALL)
-                    .border_type(BorderType::Thick),
-                area,
-            );
-
-            let chunk = Layout::default()
-                .vertical_margin(1)
-                .horizontal_margin(1)
-                .direction(Direction::Vertical)
-                .constraints(
-                    [Constraint::Min(1), Constraint::Length(1)]
-                        .as_ref(),
-                )
-                .split(area)[0];
-            f.render_widget(
                 Paragraph::new(
                     self.get_text(&self.theme, area.width)?,
                 )
+                .block(
+                    Block::default()
+                        .title(strings::SELECT_BRANCH_POPUP_MSG)
+                        .borders(Borders::ALL)
+                        .border_type(BorderType::Thick),
+                )
                 .scroll((scroll, 0))
                 .alignment(Alignment::Left),
-                chunk,
+                area,
             );
         }
 

--- a/src/ui/mod.rs
+++ b/src/ui/mod.rs
@@ -53,7 +53,7 @@ pub fn centered_rect(
         .split(popup_layout[1])[1]
 }
 
-/// makes sure Rect `r` at least stays as big as `width`/`height`
+/// makes sure Rect `r` at least stays as big as `width` & `height`
 pub fn rect_min(width: u16, height: u16, r: Rect) -> Rect {
     let new_width = r.width.max(width);
     let new_height = r.height.max(height);

